File: MVPJava/app/src/main/java/com/vascome/fogtail/appmodules/EntityTypeAdapterFactory.java
package com.vascome.fogtail.appmodules;

import com.google.gson.TypeAdapterFactory;
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory;

/**
 * Created by vasilypopov on 11/22/17
 * Copyright (c) 2017 MVPJava. All rights reserved.
 */

@GsonTypeAdapterFactory
public abstract class EntityTypeAdapterFactory implements TypeAdapterFactory {

    public static TypeAdapterFactory create() {
        return new AutoValueGson_EntityTypeAdapterFactory();
    }
}

File: MVVMKotlin/app/src/main/java/com/vascome/fogtail/data/network/EntityTypeAdapterFactory.kt
package com.vascome.fogtail.data.network

import com.google.gson.TypeAdapterFactory
import com.ryanharter.auto.value.gson.GsonTypeAdapterFactory

/**
 * Created by vasilypopov on 11/22/17
 * Copyright (c) 2017 MVPJava. All rights reserved.
 */

@GsonTypeAdapterFactory
abstract class EntityTypeAdapterFactory : TypeAdapterFactory {
    companion object {

        @JvmStatic fun create(): TypeAdapterFactory {
            return AutoValueGson_EntityTypeAdapterFactory()
        }
    }
}

